File: c_sharp/day_four.cs
namespace day_four{
    public static class day_four{
        public static int overlapping_assignments(string readPath){
            var result = 0;
            var lines = File.ReadLines(readPath);

            foreach (var line in lines){
                var eleves = line.Split(',');

                var elf_one = eleves[0].Split('-');
                var start_one = int.Parse(elf_one[0]);
                var end_one = int.Parse(elf_one[1]);

                var elf_two = eleves[1].Split('-');
                var start_two = int.Parse(elf_two[0]);
                var end_two = int.Parse(elf_two[1]);

                if (start_one == start_two) result++;
                else if (start_one > start_two) { if (end_two >= start_one) result++; }
                else if (end_one >= start_two) result++;
            }

            return result;
        }
    }
}

File: rust/src/day_four.rs
use std::{fs, env};

pub fn overlapping_assignments() -> usize{
    let args: Vec<String> = env::args().collect();

    let read_path: &String = &args[1];

    let contents = fs::read_to_string(read_path).expect(
        &format!("Should have been able to read file from {}", read_path)
        );

    contents.lines().into_iter().filter(|s|{
        let elves: Vec<&str> = s.split(',').collect();

        let elf_one: Vec<&str> = elves[0].split('-').collect();
        let start_one = elf_one[0].parse::<u32>().unwrap();
        let end_one = elf_one[1].parse::<u32>().unwrap();

        let elf_two: Vec<&str> = elves[1].split('-').collect();
        let start_two = elf_two[0].parse::<u32>().unwrap();
        let end_two = elf_two[1].parse::<u32>().unwrap();

        start_one == start_two
        || (start_one > start_two && end_one <= end_two)
        || (start_one < start_two && end_one >= end_two)
    }).count()
}
